### PR TITLE
7 post comments

### PR DESCRIPTION
**Pull request by elifalinli**

### CODE REVIEW

File: app.test.js [hunk 1]
```diff
@@ -147,3 +147,40 @@ describe("/api/reviews/:review_id/comments", () => {
...
+          review_id: 5,
+          author: "philippaclaire9",
+          body: "cool game!",
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
+  it("GET 400: should respond with an error message indicating requested id is invalid. ", () => {
+    return request(app)
```
Comment (NRMSMITH) [inline app.test.js:172]: you need to send something on this as well to prove that it's definitely testing the path

File: app.test.js [hunk 2]
```diff
@@ -147,3 +147,40 @@ describe("/api/reviews/:review_id/comments", () => {
...
+  it("GET 400: should respond with an error message if the username is not valid. ", () => {
+    const newComment = { username: "elif", body: "cool game!" };
+    return request(app)
+      .post("/api/reviews/5/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("The key is not present!");
+      });
+  });
```
Comment (NRMSMITH) [inline app.test.js:231]: A couple more tests to add: 
201 or 400 (up to you): ignores unnecessary properties on the sent comment
404: non existent id
400: missing required keys

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
your code is excellent -just a couple more tests to implement - see my comments. Please send another pr when done :)

File: app.test.js [hunk 3]
```diff
@@ -147,3 +147,83 @@ describe("/api/reviews/:review_id/comments", () => {
...
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  it("GET 400: should respond with an error message if the username is not valid. ", () => {
+    const newComment = { username: "elif", body: "cool game!" };
+    return request(app)
+      .post("/api/reviews/5/comments")
+      .send(newComment)
+      .expect(400)
```
Comment (popatre) [inline app.test.js]: This should be a 404 - username not found, please.

File: app.test.js [hunk 4]
```diff
@@ -147,3 +147,83 @@ describe("/api/reviews/:review_id/comments", () => {
...
+  });
+  it("GET 400: should respond with an error message if there is a missing key in the request. ", () => {
+    const newComment = { username: "philippaclaire9" };
+    return request(app)
+      .post("/api/reviews/5/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("something is missing with your request!");
+      });
```
Comment (popatre) [inline app.test.js:229]: Looks good on the tests - just need to change the username one to be a 404

File: controllers/error-handling.controllers.js [hunk 5]
```diff
@@ -1,20 +1,34 @@
...
\ No newline at end of file
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad Request" });
+  } else {
+    next(err);
+  }
+};
+exports.handleForeignKeyErrors = (err, req, res, next) => {
+  if (err.code === "23503") {
+    res.status(400).send({ msg: "something is wrong with your request!" });
```
Comment (popatre) [inline controllers/error-handling.controllers.js]: Im guessing you'll just have to change this one to a 404 - username not found to get the test to pass.

File: controllers/games.controllers.js [hunk 6]
```diff
@@ -47,3 +48,14 @@ exports.getCommentsByReviewId = (req, res, next) => {
...
 };
+
+exports.postComment = (req,res,next) => {
+  const newComment = req.body;
+  const id = req.params.review_id
+  insertComment(newComment, id).then((comment) => {
+    res.status(201).send({comment: comment})
+  })
+  .catch((err) => {
+    next(err);
```
Comment (popatre) [inline controllers/games.controllers.js]: Lovely stuff

**Review by popatre — CHANGES_REQUESTED** [review]
Good stuff!
Just need to change one of the tests to be a 404 username not found, instead of 400

File: controllers/games.controllers.js [hunk 7]
```diff
@@ -56,6 +56,7 @@ exports.postComment = (req,res,next) => {
     res.status(201).send({comment: comment})
   })
   .catch((err) => {
+    console.log(err)
```
Comment (popatre) [inline controllers/games.controllers.js:59]: just a console log to remove

**Review by popatre — APPROVED** [review]

**Pull request merged**